triad/utils/pyarrow.py

```diff
@@ -403,22 +403,23 @@ def _type_to_expression(dt: pa.DataType) -> str:
     raise NotImplementedError(f"{dt} is not supported")
 
 
+def _parse_types(v: Any):
+    if isinstance(v, str):
+        return _parse_type(v)
+    elif isinstance(v, Dict):
+        return pa.struct(_construct_struct(v))
+    elif isinstance(v, List):
+        assert_or_throw(1 == len(v), SyntaxError(f"{v} is not a valid list type"))
+        return pa.list_(_parse_types(v[0]))
+    else:
```

Comment:
I suggest you add back `# pragma: no cover` to bring the coverage to 100% for now. Alternatively, you can add a test to cover the scenario.

Reply:
I tried adding a test to cover the scenario, but I'm actually not sure how to reach it. 😅 

Is this expected?

```python
>>> from triad.utils.pyarrow import expression_to_schema
>>> expression_to_schema("a:1")
Traceback (most recent call last):
  File "/Users/deepyaman/triad/triad/utils/pyarrow.py", line 138, in expression_to_schema
    return pa.schema(_construct_struct(obj))
  File "pyarrow/types.pxi", line 3242, in pyarrow.lib.schema
  File "/Users/deepyaman/triad/triad/utils/pyarrow.py", line 422, in _construct_struct
    yield pa.field(k, _parse_types(v))
  File "/Users/deepyaman/triad/triad/utils/pyarrow.py", line 408, in _parse_types
    return _parse_type(v)
  File "/Users/deepyaman/triad/triad/utils/pyarrow.py", line 426, in _parse_type
    name, args = _parse_type_function(expr)
  File "/Users/deepyaman/triad/triad/utils/pyarrow.py", line 443, in _parse_type_function
    assert validate_column_name(name), f"Invalid expression {expr}"
AssertionError: Invalid expression 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deepyaman/triad/triad/utils/pyarrow.py", line 142, in expression_to_schema
    raise SyntaxError(f"Invalid syntax: '{expr}' {str(e)}")
SyntaxError: Invalid syntax: 'a:1' Invalid expression 1
```

I'm not sure why the key (type) should be checked with `validate_column_name`.

Reply:
The naming of the private function needs improvement. I double checked the logic, it is correct.